zookeeper-jute/src/main/java/org/apache/jute/compiler/JRecord.java

```diff
@@ -767,4 +795,127 @@ public static String getCsharpFQName(String name) {
         }
         return fQName.toString();
     }
+
+    public String getJavaFieldComments(JField jField) {
+        return getFieldComments(jField, "  ");
+    }
+
+    public String getCFieldComments(JField jField) {
+        return getFieldComments(jField, "    ");
+    }
+
+    private String getFieldComments(JField jField, String indent) {
+        if (jField == null || jField.getTypeToken() == null || jField.getNextToken() == null) {
+            return "";
+        }
+
+        // get the comment before the line
+        List<String> comments = extractLeadingComments(getCommentToken(jField.getTypeToken().specialToken));
+
+        // get the end-of-line comments of fields
+        // If the current field and the next field are on the same line,
+        // the leading field comment of the next field should be discarded
+        Token commentToken = getCommentToken(jField.getNextToken().specialToken);
+        if (commentToken != null && jField.getTypeToken().beginLine == commentToken.beginLine) {
+
+            if (jField.getNextToken().specialToken == commentToken) {
+                jField.getNextToken().specialToken = null;
+            }
+
+            if (commentToken.next != null) {
+                commentToken.next.specialToken = null;
+                commentToken.next = null;
```

Comment:
I am thinking whether we should have a `previousToken`. This way `getCommentToken` can skip end-line comment for previous entry without modifying things in code generation. I expect code generation is reproducible.

Reply:
Also `extractLeadingComments` could take a `sentinelToken` to limit end-line comment traversal without changing `Token.next` field.